### DISCUSSION TITLE
Do I need clearTag  to render style tag in RSC?

### DISCUSSION THREAD

**Discussion by dante01yoon** [styled-components/styled-components#4059] [Q&A] [answered]
Greetings.
In Nextjs docs, There is  examples codes for beginners who using with  Nextjs 13 app directory

```tsx
'use client'
 
import React, { useState } from 'react'
import { useServerInsertedHTML } from 'next/navigation'
import { ServerStyleSheet, StyleSheetManager } from 'styled-components'
 
export default function StyledComponentsRegistry({
  children,
}: {
  children: React.ReactNode
}) {
  // Only create stylesheet once with lazy initial state
  // x-ref: https://reactjs.org/docs/hooks-reference.html#lazy-initial-state
  const [styledComponentsStyleSheet] = useState(() => new ServerStyleSheet())
 
  useServerInsertedHTML(() => {
    const styles = styledComponentsStyleSheet.getStyleElement()
    styledComponentsStyleSheet.instance.clearTag()
    return <>{styles}</>
  })
 
  if (typeof window !== 'undefined') return <>{children}</>
 
  return (
    <StyleSheetManager sheet={styledComponentsStyleSheet.instance}>
      {children}
    </StyleSheetManager>
  )
}
```

I can't find what exactly `styledComponentsStyleSheet.instance.clearTag()` does. Should I using with in React server component?

**Comment by crs1138** [comment]
Also, what would the equivalent be for version of Styled Components below 6? As the `clearTag()` function does not exists and setting the property `instance.tag = undefined` doesn't work.

**Comment by quantizor** [comment, marked as answer]
The upcoming release overhauls RSC style injection — styled components in server components now emit plain inline `<style>` tags directly, so the `clearTag` / `useServerInsertedHTML` registry pattern is no longer the only option for RSC.

You can test the prerelease:

```
npm install styled-components@test
```

See the staged release PR for details: https://github.com/styled-components/styled-components/pull/5687

For the App Router client component registry pattern (with `useServerInsertedHTML`), `clearTag()` is still used — but server components themselves no longer need it.
